### PR TITLE
Remove Compositor extra dims using composite_dims

### PR DESCRIPTION
**Pull request by jmilloy**
Adds `composite_dims` trait and removes extra dimensions from the evaluation coordinates.

 - [x] autodetect, ignore, or fail if the `composite_dims` are not provided?
 - [x] tests

Autodetection could look like this at its simplest:

```
@tl.default("composite_dims")
    def _default_composite_dims(self):
        return self.sources[0].coordinates.udims
```

### CODE REVIEW

**Comment by mpu-creare** [comment]
I'm a little hesitant about the autodetection because of all the circular reference we run into whenever we have a default set from `compositor.sources`. Tagging `dims` as an attribute makes this fixable in any pipeline, so I'm happy with this. Works for my current case in any case. 

Just needs tests then I'm happy to merge.

File: podpac/core/compositor/compositor.py [hunk 1]
```diff
@@ -52,6 +53,7 @@ class BaseCompositor(Node):
     interpolation = InterpolationTrait(allow_none=True, default_value=None).tag(attr=True)
     source_coordinates = tl.Instance(Coordinates, allow_none=True, default_value=None).tag(attr=True)
 
+    composite_dims = tl.List(trait=Dimension())
```
[the remark this answers is not in the document]
Comment (jmilloy) [reply podpac/core/compositor/compositor.py]: Yeah, I had the same thought.

**Comment by jmilloy** [comment]
Note

The obvious way to remove extra dimensions would be something like this:

```
extra = [dim for dim in coordinates.udims if dim not in self.dims]
coordinates = coordinates.drop(extra)
```

But if the compositor dims are `lat, lon` and the requested dims are `lat, lon, time_alt`, you have to remove the `time_alt` dimension together. On the other hand, if the requested dims are `lat_lon_alt`, you don't need to remove alt at all. So the removal is implemented like this:

```
extra = [
    c.name
    for c in coordinates.values()
    if (isinstance(c, Coordinates1d) and c.name not in self.dims)
    or (isinstance(c, StackedCoordinates) and all(dim not in self.dims for dim in c.dims))
]
coordinates = coordinates.drop(extra)
```

It is also implemented this way in `DataSource.eval`. In both the Compositor and DataSource tests, the `lat, lon, alt_time` case is tested, but the `lat_lon_time` case is not tested (commented out) because the interpolator cannot handle this case yet.

File: podpac/core/compositor/ordered_compositor.py [hunk 2]
```diff
@@ -50,7 +50,15 @@ def composite(self, coordinates, data_arrays, result=None):
         mask = UnitsDataArray.create(coordinates, outputs=self.outputs, data=0, dtype=bool)
         for data in data_arrays:
             if self.outputs is None:
-                data = data.transpose(*result.dims)
+                try:
+                    data = data.transpose(*result.dims)
+                except ValueError:
+                    raise NodeException(
```
Comment (mpu-creare) [inline podpac/core/compositor/ordered_compositor.py:56]: Superb. Nice addition.

**Comment by mlshapiro** [comment]
FYI I merged in my changes to Terrain Tiles and the TileCompositor here since it would conflict as a separate branch.  See bc2bb163776f1cdeace8ef4075fe027160b57ef7 for more information. Long story short, Terrain Tiles should now composite correctly. Whew. Thanks @jmilloy

**Pull request merged**